src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedItems: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const value = String(item[key]);
+
+    if (!groupedItems[value]) {
+      groupedItems[value] = [item];
+    } else {
+      groupedItems[value].push(item);
+    }
```

Comment:
You can also create new array if statement is true and call push method without else